pkg/api/dashboard.go

```diff
@@ -394,14 +396,18 @@ func (hs *HTTPServer) postDashboard(c *models.ReqContext, cmd models.SaveDashboa
 	cmd.OrgId = c.OrgID
 	cmd.UserId = c.UserID
 	if cmd.FolderUid != "" {
-		folder, err := hs.folderService.GetFolderByUID(ctx, c.SignedInUser, c.OrgID, cmd.FolderUid)
+		ctx := appcontext.WithUser(c.Req.Context(), c.SignedInUser)
```

Comment:
The `c.Req.Context()` will already have the signed in user set; that's handled by the [contextHandler middleware](https://github.com/grafana/grafana/blob/main/pkg/services/contexthandler/contexthandler.go#L101)

We should instead pass the entire `c.Req.Context` as the context for the folder service; that's exactly what it's there for :)